### ISSUE TITLE
no-nested-ternary doesn't always work

### ISSUE THREAD

**Issue by kaiyoma** [eslint/eslint#12641]
**Tell us about your environment**

```
Node version: v10.16.0
npm version: v6.13.0
Local ESLint version: v6.6.0 (Currently used)
```

**What parser (default, Babel-ESLint, etc.) are you using?**
TypeScript and Babel.

**What did you do?**

```js
const buttonStyle = isTextButton ? (isRisk ? 'danger' : props.type) || 'default' : 'link';
```

**What did you expect to happen?**
I expect the `no-nested-ternary` rule to be activated and to throw an error.

**What actually happened? Please include the actual, raw output from ESLint.**
No error was thrown. However, if I remove the `|| 'default'` portion of the line above, eslint correctly prints an error for this line of code:

```js
const buttonStyle = isTextButton ? (isRisk ? 'danger' : props.type) : 'link';
```

It would appear that the rule is looking for *exactly* a nested ternary and any other bit of code causes the rule to no longer work correctly.

**Comment by mdjermanovic** [comment]
Hi @kaiyoma, thanks for the issue!

As you already noticed, the rule targets only directly nested ternary expressions.

This is by design, i.e., works as intended, but an additional option to also target deeply nested ternary expression seems reasonable to me. 

You can open a rule enhancement proposal for an additional option (there was already a proposal in #10352, but feel free to open a new one to re-evaluate).

This can be also enforced using [`no-restricted-syntax`](https://eslint.org/docs/rules/no-restricted-syntax) as described in [#12170 (comment)](https://github.com/eslint/eslint/issues/12170#issuecomment-524889787), albeit with some limitations.
